lint/stylelint.bzl

```diff
@@ -0,0 +1,208 @@
+"""Configures [Stylelint](https://stylelint.io/) to run as a Bazel aspect
+
+First, all CSS sources must be the srcs of some Bazel rule.
+You can use a `filegroup` with `lint-with-stylelint` in the `tags`:
+
+```python
+filegroup(
+    name = "css",
+    srcs = glob(["*.css"]),
+    tags = ["lint-with-stylelint"],
+)
+```
+
+See the `filegroup_tags` and `rule_kinds` attributes below to customize this behavior.
+
+## Usage
+
+Add `stylelint` as a `devDependency` in your `package.json`, and declare a binary target for Bazel to execute it.
+
+For example in `tools/lint/BUILD.bazel`:
+
+```starlark
+load("@npm//:stylelint/package_json.bzl", stylelint_bin = "bin")
+stylelint_bin.stylelint_binary(name = "stylelint")
+```
+
+Then declare the linter aspect, typically in `tools/lint/linters.bzl`:
+
+```starlark
+load("@aspect_rules_lint//lint:stylelint.bzl", "lint_stylelint_aspect")
+stylelint = lint_stylelint_aspect(
+    binary = "@@//tools/lint:stylelint",
+    config = "@@//:stylelintrc",
+)
+```
+
+Finally, register the aspect with your linting workflow, such as in `.aspect/cli/config.yaml` for `aspect lint`.
+"""
+
+load("@aspect_bazel_lib//lib:copy_to_bin.bzl", "COPY_FILE_TO_BIN_TOOLCHAINS", "copy_files_to_bin_actions")
+load("//lint/private:lint_aspect.bzl", "LintOptionsInfo", "filter_srcs", "output_files", "patch_and_output_files", "should_visit")
+
+_MNEMONIC = "AspectRulesLintStylelint"
+
+def stylelint_action(ctx, executable, srcs, config, stderr, exit_code = None, env = {}, options = []):
+    """Spawn stylelint as a Bazel action
+
+    Args:
+        ctx: an action context OR aspect context
+        executable: struct with an _stylelint field
+        srcs: list of file objects to lint
+        config: js_library representing the config file (and its dependencies)
+        stderr: output file containing the stderr or --output-file of stylelint
+        exit_code: output file containing the exit code of stylelint.
+            If None, then fail the build when eslint exits non-zero.
```

Comment:
This says "eslint", should it be Stylelint?

Reply:
thanks, https://github.com/aspect-build/rules_lint/pull/345